packages/sw-cache-expiration/src/lib/behavior.js

```diff
@@ -14,144 +14,257 @@
 */
 
 import idb from 'idb';
-import Configuration from './configuration';
 import assert from '../../../../lib/assert';
-import {idbName, idbVersion} from './constants';
+import {idbName, idbVersion, urlProperty, timestampProperty} from './constants';
 
 /**
- * TODO Behavior Description
  * @memberof module:sw-cache-expiration
+ *
+ * @example
+ * // Used as an automatically invoked as "behavior" by a RequestWrapper:
+ *
+ * const cacheName = 'runtime-cache';
+ * const requestWrapper = new goog.runtimeCaching.RequestWrapper({
+ *   cacheName,
+ *   behaviors: [
+ *     new goog.cacheExpiration.Behavior({cacheName, maxEntries: 10})
+ *   ]
+ * });
+ *
+ * // Set up a route to match any requests made against the example.com domain.
+ * // The requests will be handled with a stale-while-revalidate policy, and the
+ * // cache size will be capped at 10 entries.
+ * const route = new goog.routing.RegExpRoute({
+ *   match: ({url}) => url.domain === 'example.com',
+ *   handler: new goog.runtimeCaching.StaleWhileRevalidate({requestWrapper})
+ * });
+ *
+ * @example
+ * // Explicitly invoked usage independent of the goog.routing framework, via
+ * // the expireOldEntries() method:
+ *
+ * // TODO: Write sample code.
  */
 class Behavior {
   /**
-   * @param {Object} input
-   * @param {Object} input.configuration
+   * Creates a new `Behavior` instance, which is used to remove entries from a
+   * [`Cache`](https://developer.mozilla.org/en-US/docs/Web/API/Cache) once
+   * certain criteria—maximum number of entries, age of entry, or both—is met.
+   *
+   * @param {Object} input The input object to this function.
+   * @param {string} input.cacheName The name of the cache.
+   * @param {Number} [input.maxEntries] The maximum size of the cache. Entries
+   *        will be expired using a LRU policy once the cache reaches this size.
+   * @param {Number} [input.maxAgeSeconds] The maximum age for fresh entries.
    */
-  constructor({configuration}) {
-    if (typeof configuration === 'undefined') {
-      configuration = {};
-    }
+  constructor({cacheName, maxEntries, maxAgeSeconds} = {}) {
+    assert.isType({cacheName}, 'string');
+    assert.atLeastOne({maxEntries, maxAgeSeconds});
 
-    assert.isInstance({configuration}, Configuration);
-    this.configuration = configuration;
+    this.cacheName = cacheName;
+    this.maxEntries = maxEntries;
+    this.maxAgeSeconds = maxAgeSeconds;
   }
 
   /**
-   * Get the cache name
-   * @memberof module:sw-cache-expiration.Behavior
-   * @member {String} cacheName
+   * Returns a promise for the IndexedDB database used to keep track of state
+   * for `this.cacheName`.
+   *
+   * @private
+   * @return {DB} An open DB instance.
    */
-  get cacheName() {
-    return this.configuration.cacheName;
-  }
+  async getDB() {
```

Comment:
Yup, basically.

Longer term, it might make sense to move the other libraries from `indexedDBHelper` (which dates back to the Appcache helper library) to `idb` or `idb-keyval` directly, given the additional flexibility they offer.